modules/sharedIdSystem.js

```diff
@@ -0,0 +1,127 @@
+/**
+ * This module adds Shared ID support to the User ID module
+ * The {@link module:modules/userId} module is required.
+ * @module modules/sharedIdSystem
+ * @requires module:modules/userId
+ */
+import { submodule } from '../src/hook.js';
+import * as utils from '../src/utils.js';
+// These values should NEVER change. If
+// they do, we're no longer making ulids!
+const ENCODING = '0123456789ABCDEFGHJKMNPQRSTVWXYZ'; // Crockford's Base32
+const ENCODING_LEN = ENCODING.length;
+const TIME_MAX = Math.pow(2, 48) - 1;
+const TIME_LEN = 10;
+const RANDOM_LEN = 16;
+const MODULE_NAME = 'sharedId';
+
+/** @type {Submodule} */
+export const sharedIdSubmodule = {
+  /**
+   * used to link submodule with config
+   * @type {string}
+   */
+  name: MODULE_NAME,
+
+  sharedIdFactory: undefined,
+
+  /**
+   * decode the stored id value for passing to bid requests
+   * @function
+   * @param {string} value
+   * @returns {{sharedId:string}}
+   */
+  decode(value) {
+    return (value && typeof value['sharedId'] === 'string') ? { 'sharedId': value['sharedId'] } : undefined;
+  },
+
+  /**
+   * performs action to obtain id and return a value.
+   * @function
+   * @param {SubmoduleParams} [configParams]
+   * @returns {sharedId}
+   */
+  getId: function(configParams) {
+    let sharedId = this.sharedIdGenerator();
+    return {
+      id: sharedId }
+  },
+
+  sharedIdGenerator: function () {
+    if (!this.sharedIdFactory) {
+      this.sharedIdFactory = this.factory();
+    }
+
+    return this.sharedIdFactory();
+  },
+
+  factory: function (currPrng) {
```

Comment:
I had to google what `Prng` stands for, would be good to indicate what that is here. A docBlock would solve for this.